### PR TITLE
Wait for AVAILABLE status before returning from create

### PR DESCRIPTION
**Pull request by garthy**

### CODE REVIEW

File: pkg/provisioner/block/block.go [hunk 1]
```diff
@@ -80,6 +87,70 @@ func roundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int64 {
...
+		}
+
+		if getVolumeResponse.LifecycleState == core.VolumeLifecycleStateTerminating {
+			return false, fmt.Errorf("Failed to provision volume. Volume: %v is in Terminating state", *volumeID)
+		}
+
+		return false, nil
+	}
+
+	ready, err := isVolumeReady()
```
Comment (prydie) [inline pkg/provisioner/block/block.go]:
```go
return wait.PollImmediate(time.Second * 5, timeout, func() (bool, error) {
 	ready, err := isVolumeReady() 
	if err != nil {
		return false, fmt.Errorf("failed to provision volume %q: %v", *volumeID, err)
	}
	return ready, nil
})
```

`k8s.io/apimachinery/pkg/util/wait`

File: pkg/provisioner/block/block.go [hunk 2]
```diff
@@ -132,6 +203,18 @@ func (block *blockProvisioner) Provision(options controller.VolumeOptions, ad *i
...
 	}
 
+	err = block.waitForVolumeAvailable(newVolume.Id, block.timeout)
+	if err != nil {
+		// Delete the volume if it failed to get in a good state for us
+		request := core.DeleteVolumeRequest{VolumeId: newVolume.Id}
+		ctx, cancel := context.WithTimeout(block.client.Context(), block.client.Timeout())
+		defer cancel()
+
+		_, _ = block.client.BlockStorage().DeleteVolume(ctx, request)
```
Comment (prydie) [inline pkg/provisioner/block/block.go]:
```go
_, _ = block.client.BlockStorage().DeleteVolume(ctx, core.DeleteVolumeRequest{
	VolumeId: newVolume.Id,
})

File: pkg/provisioner/block/block_test.go [hunk 3]
```diff
@@ -102,8 +108,9 @@ func (p *mockProvisionerClient) TenancyOCID() string {
 }
 
 // NewClientProvisioner creates an OCI client from the given configuration.
-func NewClientProvisioner(pcData client.ProvisionerClient) client.ProvisionerClient {
-	return &mockProvisionerClient{}
+func NewClientProvisioner(pcData client.ProvisionerClient,
```
Comment (prydie) [inline pkg/provisioner/block/block_test.go]: Either

```go
func NewClientProvisioner(pcData client.ProvisionerClient, storage *mockBlockStorageClient) client.ProvisionerClient {
```

or 

```go
func NewClientProvisioner(
	pcData client.ProvisionerClient,
	storage *mockBlockStorageClient,
) client.ProvisionerClient {
```

File: pkg/provisioner/block/block.go (hunk 1 above)
Comment (garthy) [reply pkg/provisioner/block/block.go]: I was halfway through writing this and then thought K8s would have something...

**Pull request merged**